frontend/src/search.js

```diff
@@ -83,6 +83,34 @@ search.addWidgets([
         document.getElementById('under-progress').classList.remove('hidden');
     },
   },
+  {
+    message: '',
+    init() {
+      document.getElementById('retry-search').addEventListener('click', () => {
+        document.getElementById('loading').classList.remove('hidden');
+        document.getElementById('stats').classList.add('hidden');
+        document.getElementById('error').classList.add('hidden');
+        this.message = '';
+        setTimeout(() => {
+          search.helper.search();
```

Comment:
Here and below we are referring the global variable `search`, which makes the widget not isolated. Can't we get the `helper` reference from the parameters of the `init` method or any other way natural for a widget?

Reply:
[Documentation](https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-display/js/#handling-errors) shows we can access the helper in the `init` method but hooking into the error event through the helper does not work as expected.

Reply:
Very relevant link! I regret I had not found it in the first place. It's exactly about what we want to achieve — error-handling widget.

Can you explain in details what you mean by “does not work as expected”? How does it work?

Reply:
Thanks, I confirm that the error handler does not get triggered. I've posted a question on the official forum: https://discourse.algolia.com/t/error-handling-widget/13384. Let's wait for a day or two.